File: TOKI/Course_Pemrograman Dasar/08 Analisis Kompleksitas/B.py
import math

def cekPrima(n):
    if n == 1:
        return False
    elif n == 2:
        return True
    else:
        y = int(math.sqrt(n))
        test = True
        while y >= 2 and test:
            if n % y == 0:
                test = False
            y -= 1
        return test

    # jumlah = 0
    # for i in range(1, n+1):
    #     if n%i == 0:
    #         jumlah += 1
    
    # return (jumlah == 2)

n = int(input())

array = []
for _ in range(n):
    array.append(int(input()))

for angka in array:
    if cekPrima(angka):
        print("YA")
    else:
        print("BUKAN")

File: TOKI/Course_Pemrograman Dasar/08 Analisis Kompleksitas/B.cpp
#include <cstdio>
#include <cmath>

using std::sqrt;

int main() {
    int n;
    scanf("%d", &n);
    
    int angka[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &angka[i]);
    }
    

    for (int i = 0; i < n; i++) {
        if (angka[i] <= 1) {
            printf("BUKAN\n");
        } else {
            bool prima = true;
            for (int j = 2; j <= std::sqrt(angka[i]); j++) {
                if (angka[i] % j == 0) {
                    prima = false;
                    break;
                }
            }
            if (prima) {
                printf("YA\n");
            } else {
                printf("BUKAN\n");
            }
        }
    }
     
    return 0;
}